### DISCUSSION TITLE
Why no one have discussion in here? let's start to talk about SSL and Proxy

### DISCUSSION THREAD

**Discussion by chongy076** [ParisNeo/lollms-webui#138] [General]
Hi All,

1. SSL features , upcoming. do give us sometimes to test. Since SSL involved the whole programs.
2. Proxy , may add field in config.  

Best Regards

**Comment by ParisNeo** [comment]
Hi, yes it is an issue. I had no time to work on this. But I am open to contributions :)

**Reply by chongy076** [reply]
Hope , we can learn until 90 years old.

hahaha.

**Reply by chongy076** [reply]
I supposed to rest, I just recovered from sick. But I can't help it, while I was sick. I stumble with gpt4all and posted to you on the suggestion.  
This gpt got so much potential and really made us feel like living in future.

**Reply by chongy076** [reply]
I had reinstalled it , but i think is the model i used need extra control (for now just ignore it first), otherwise it would have non stop generating the sample response. for SSL , Just now i tested, yes , i do need to update all the *.js. Also , after the pull, the client side also need to changed so that the client socketio  not disconnected prematurely. I supposed to work on that after the first fork pull to match with server end.

I also need to check the transmitted data , so that it is correct encrypted. 

![image](https://user-images.githubusercontent.com/50088438/235627208-401e5d3d-acc7-47aa-b3ce-a74bdf1d2614.png)

**Comment by chongy076** [comment]
it seem i missed all the fun, but i am having eye's problem. still in the mid of recovering. 

oogabooga , I had seen their code and tested. have to said it  had many issues to get it to run due to the max mem size for certain computer and GPU. yes, they had GPU and non GPU one, but some pretty good.

I may need pause for a moment until my eye's are recovered. Sorry for the inconvenient. It came at the wrong timing.

but the code i had placed in the previous issues just as a back up. if anyone wanted to have it early.

but front  .js need  to change and attention needed to add that https and proxy.

if we didn't use socketio, the fast way will be [https://pypi.org/project/waitress/](url).

this would be easy for small or medium size request, since we are using socketio we have to use below code.

[this is before the latest backend changed]
[app.txt](https://github.com/nomic-ai/gpt4all-ui/files/11403559/app.txt)
[https_main.txt](https://github.com/nomic-ai/gpt4all-ui/files/11403562/https_main.txt)

**Comment by ParisNeo** [comment]
Thank you man,

Take care of yourself.
We are working on a new very cool ui. And we hope to fix all those issues soon.

Contributions are all welcome here. There are too many ideas but we need to focus in order to bring a good beatutiful simple to use tool.
And yes, being able to simply support many backends is a good idea, but in my vision, we should separate them in different backends that the people may install and uninstall them as they whish.

Wish you fast recovery.
